Assets/Scripts/UI/Pickup Pup/Currency/CurrencyDisplay.cs

```diff
@@ -5,19 +5,68 @@
 
 public class CurrencyDisplay : PPUIElement 
 {
-    CurrencyData currency;
-
-    // Sets the Currency to display
-    public void SetCurrency(CurrencyData currency)
-    {
-        this.currency = currency;
-        OnUpdate();
-    }
-
-    // Updates text to show the new amount of Currency
-    public void OnUpdate()
-    {
-        text.text = currency.Amount.ToString();
-    }    
+	CurrencyType type;
+	PPDataController dataController;
+
+	public void Init(PPDataController dataController, CurrencyType type)
```

Comment:
Yes, but I like use Dependency Injection when possible (https://en.wikipedia.org/wiki/Dependency_injection). I think it makes the code cleaner